src/components/views/AppSettingsView.tsx

```diff
@@ -246,169 +269,41 @@ class AppSettingsView extends React.Component<Props, State> {
             </VisibleWithRole>
           </AccordionDetails>
         </Accordion>
+        { this.renderDeleteApplicationDialog() }
       </>
     );
   }
 
   /**
-   * Handles importing data from wall json file
+   * Render delete application confirmation dialog
    */
-  private handleWallJsonImport = async (files: FileList | null) => {
-    if (!files) {
-      return;
-    }
-
-    const file = files.item(0);
-    if (!file) {
-      return;
-    }
-
-    const { rootResource } = this.props;
-    const reader = new FileReader();
-    reader.onload = async e => {
-      if (!e.target) {
-        return;
-      }
-      const data = JSON.parse(e.target.result as string)
-      const topLevel = data.root.children;
-
-      this.setState({ importingContent: true });
-
-      const imported = await this.importWallJsonItems(rootResource.id!, topLevel);
-      if (imported) {
-        await this.importRootProperties(data);
-      }
-
-      this.setState({ importDone: imported });
-
-      toast.success(strings.importDone);
+  private renderDeleteApplicationDialog = () => {
+    const { onDeleteApplicationClick, application  } = this.props;
+    const { deleteApplicationDialogOpen } = this.state;
 
-      setTimeout(() => window.location.reload(), 3000);
-    }
-    reader.readAsText(file);
+    return (
+      <GenericDialog
+        title={ strings.applicationSettings.deleteApplication }
+        onCancel={ this.toggleDeleteApplicationDialog }
+        onClose={ this.toggleDeleteApplicationDialog }
+        onConfirm={ () => onDeleteApplicationClick(application) }
```

Comment:
move this to private method, you also do not need to invoke the method here because application is available to component via props

Reply:
onDeleteApplication comes directly from props and it is more organized to send the actual application as a parameter